soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -658,42 +598,119 @@ void CheckTrackerItemReceive(GetItemEntry giEntry) {
+    switch (flagType) {
+        case FLAG_GS_TOKEN:
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_GOLD_SKULLTULA;
+            break;
+        case FLAG_EVENT_CHECK_INF:
+            if (flag == 0x0F) {
+                SetCheckCollected(RC_HIDEOUT_GERUDO_MEMBERSHIP_CARD);
+                return;
+            }
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_EVENT_CHK_INF;
+            break;
+        case FLAG_INF_TABLE:
+            if (flag == 400) {
+                SetCheckCollected(RC_GF_HBA_1000_POINTS);
+            } else if (flag == 286) {
+                SetCheckCollected(RC_GC_ROLLING_GORON_AS_CHILD);
+            } else if (flag == 265) {
+                SetCheckCollected(RC_GC_ROLLING_GORON_AS_ADULT);
+            } else if (flag == 313) {
+                SetCheckCollected(RC_ZD_KING_ZORA_THAWED);
+            } else if (flag == 401) {
+                SetCheckCollected(RC_MARKET_LOST_DOG);
+            }
+            if (!IS_RANDO) {
+                if (flag == 402) {
+                    SetCheckCollected(RC_LW_DEKU_SCRUB_NEAR_BRIDGE);
+                } else if (flag == 403) {
+                    SetCheckCollected(RC_LW_DEKU_SCRUB_GROTTO_FRONT);
+                }
+            }
+            break;
+        case FLAG_ITEM_GET_INF:
+            if (!IS_RANDO) {
+                if (flag == 30) {
+                    SetCheckCollected(RC_DEKU_THEATER_SKULL_MASK);
+                    return;
+                } else if (flag == 31) {
+                    SetCheckCollected(RC_DEKU_THEATER_MASK_OF_TRUTH);
+                    return;
+                } else if (flag == 11) {
+                    SetCheckCollected(RC_HF_DEKU_SCRUB_GROTTO);
+                    return;
+                }
+            }
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_ITEM_GET_INF;
+            break;
+        case FLAG_RANDOMIZER_INF:
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_RANDOMIZER_INF;
+            break;
     }
-    if (gPlayState->msgCtx.msgMode != MSGMODE_NONE) {
-        checkAreas.push_back(checkArea);
-        messageCloseCheck = true;
+    if (checkMatchType == SpoilerCollectionCheckType::SPOILER_CHK_NONE) {
         return;
     }
-    if (IS_RANDO || (!IS_RANDO && giEntry.getItemCategory != ITEM_CATEGORY_JUNK)) {
-        checkAreas.push_back(checkArea);
-        checkCollected = true;
+    for (auto [rc, rcObj] : RandomizerCheckObjects::GetAllRCObjects()) {
+        if (!IS_RANDO && ((rcObj.vOrMQ == RCVORMQ_MQ && !IS_MASTER_QUEST) || (rcObj.vOrMQ == RCVORMQ_VANILLA && IS_MASTER_QUEST)) ||
+            IS_RANDO && (OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) && rcObj.vOrMQ == RCVORMQ_VANILLA) || 
+                         !OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) && rcObj.vOrMQ == RCVORMQ_MQ){
```

Comment:
The readability of this could be improved if you wrapped the two `IS_RANDO` parts in parenthesis. Also I think there is a bug in the existing parens too maybe?

```suggestion
        if ((!IS_RANDO && ((rcObj.vOrMQ == RCVORMQ_MQ && !IS_MASTER_QUEST) ||
                           (rcObj.vOrMQ == RCVORMQ_VANILLA && IS_MASTER_QUEST))) ||
            (IS_RANDO && ((OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) &&
                           rcObj.vOrMQ == RCVORMQ_VANILLA) ||
                          !OTRGlobals::Instance->gRandomizer->masterQuestDungeons.contains(rcObj.sceneId) &&
                              rcObj.vOrMQ == RCVORMQ_MQ))) {
```

Reply:
These are all done now.